tests/ops/qubit/test_special_unitary.py

```diff
@@ -341,3 +771,151 @@ def comparison(x):
 
         expected = tape.jacobian(loss, theta)
         assert np.allclose(jac, expected)
+
+
+class TestSpecialUnitaryIntegration:
+    """Test that the operation SpecialUnitary is executable and
+    differentiable in a QNode context, both with automatic differentiation
+    and parameter-shift rules."""
+
+    @staticmethod
+    def circuit(x):
+        """Test circuit involving a single SpecialUnitary operations."""
+        qml.SpecialUnitary(x, wires=[0, 1])
+        return qml.expval(qml.PauliZ(0) @ qml.PauliX(1))
+
+    x = np.linspace(0.2, 1.5, 15)
+    state = qml.SpecialUnitary.compute_matrix(x, 2) @ np.eye(4)[0]
+    exp = np.vdot(state, qml.matrix(qml.PauliZ(0) @ qml.PauliX(1)) @ state).real
+
+    def test_qnode_numpy(self):
+        """Test that the QNode executes with Numpy."""
+        dev = qml.device("default.qubit", wires=2)
+        qnode = qml.QNode(self.circuit, dev, interface=None)
+
+        res = qnode(self.x)
+        assert qml.math.shape(res) == ()
+        assert qml.math.isclose(res, self.exp)
+
+    @pytest.mark.autograd
+    def test_qnode_autograd(self):
+        """Test that the QNode executes with Autograd.
+        Neither hardware-ready nor autodiff gradients are available in Autograd."""
+
+        dev = qml.device("default.qubit", wires=2)
+        qnode = qml.QNode(self.circuit, dev, interface="autograd")
+
+        x = qml.numpy.array(self.x, requires_grad=True)
+        res = qnode(x)
+        assert qml.math.shape(res) == ()
+        assert qml.math.isclose(res, self.exp)
+
+    @pytest.mark.jax
+    @pytest.mark.parametrize("use_jit", [False, True])
+    @pytest.mark.parametrize("shots, atol", [(None, 1e-6), (10000, 1e-1)])
+    def test_qnode_jax(self, shots, atol, use_jit):
+        """Test that the QNode executes and is differentiable with JAX. The shots
+        argument controls whether autodiff or parameter-shift gradients are used."""
+        if use_jit and shots is not None:
+            pytest.skip("Hardware-ready differentiation does not support JITting yet.")
+        import jax
+
+        jax.config.update("jax_enable_x64", True)
+
+        dev = qml.device("default.qubit", wires=2, shots=shots)
+        qnode = qml.QNode(self.circuit, dev, interface="jax")
```

Comment:
Can you specify parameter-shift here? Even if it is automatically selected.

Reply:
Done.